helper/impls/helper-impl-111/src/main/java/com/linkedin/avroutil1/compatibility/avro111/codec/AliasAwareSpecificData.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+
+package com.linkedin.avroutil1.compatibility.avro111.codec;
+
+import com.linkedin.avroutil1.compatibility.ClassLoaderUtil;
+import org.apache.avro.Schema;
+import org.apache.avro.specific.SpecificData;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AliasAwareSpecificData extends SpecificData {
+
+    //same idea as the one in SpecificData
+    protected final static Map<String, Class<?>> CLASS_CACHE = new ConcurrentHashMap<>();
+    protected final static Class<?> NO_CLASS = (new Object() {}).getClass();
```

Comment:
Hmm... why not `new Object()`? Are you creating an anonymous class here? Why?

Reply:
this is a negative cache marker. its a subclass to be unique. this isnt fully. implemented yet